### PR TITLE
fix(ci): Changes to take care of the issue 8699

### PR DESCRIPTION
**Pull request by ranj8work**
Signed-off-by: Ranjini Iyer <ranj8work@gmail.com>

## Summary
Changes to take care of the issue [8699](https://github.com/magma/magma/issues/8699)

github workflows are updated to do a path filter pre-check to ensure the workflow gets executed only if the changes are made in one of the path filtered diretories associated to that component.

## Test Plan
The changes are made in cwag-workflow.yml in the master branch of my magma [forked](https://github.com/ranj8work/magma/pull/22) repo.
[PR](https://github.com/ranj8work/magma/pull/22) that is used to test the issue [8699](https://github.com/magma/magma/issues/8699) in the branch issue-8699.
Action Summary [result](https://github.com/ranj8work/magma/actions/runs/1215454284) for changes made in docs/ folder which is not one of the path filters (orc8r, lte, feg, cwf) for cwag-workflow.yml
Action Summary [result](https://github.com/ranj8work/magma/actions/runs/1215471812) for changes made in cwf/ folder which is in one of the path filters (orc8r, lte, feg, cwf) for cwag-workflow.yml

## Additional Information

- [ NO] This change is backwards-breaking

### CODE REVIEW

File: .github/workflows/agw-workflow.yml [hunk 1]
```diff
@@ -95,6 +105,9 @@ jobs:
           SLACK_FOOTER: ' '
 
   session_manager_test:
+    needs: path_filter
+    if: ${{ needs.path_filter.outputs.should_skip != 'true' }}
```
Comment (quentinDERORY) [inline .github/workflows/agw-workflow.yml]: replace with `== 'false'` everywhere you put != 'true'

File: .github/workflows/agw-workflow.yml [hunk 2]
```diff
@@ -5,18 +5,28 @@ on:  # yamllint disable-line rule:truthy
...
+      should_skip: ${{ steps.skip_check.outputs.should_skip }}
+    steps:
+      - id: skip_check
+        uses: fkirc/skip-duplicate-actions@master
+        with:
+          # Only run the main job for changes including the following paths
+          paths: '["orc8r/**", "lte/**"]'
   lte-test:
+    needs: path_filter
+    if: ${{ needs.path_filter.outputs.should_skip != 'true' }}
```
Comment (quentinDERORY) [inline .github/workflows/agw-workflow.yml]: here too

File: .github/workflows/gcc-problems.yml [hunk 3]
```diff
@@ -121,15 +139,19 @@ jobs:
         uses: actions/upload-artifact@v2
         with:
           name: build_logs_oai
-          path: ${{ github.workspace }}/compile.log      
+          path: ${{ github.workspace }}/compile.log
 
   build_session_manager:
+    needs:
+      - gen_build_container
```
Comment (quentinDERORY) [inline .github/workflows/gcc-problems.yml]: put path filter first as this will be executed first

File: .github/workflows/gcc-problems.yml [hunk 4]
```diff
@@ -64,12 +78,16 @@ jobs:
           path: /tmp/mme_builder.tar
 
   build_oai:
+    needs:
+      - gen_build_container
```
Comment (quentinDERORY) [inline .github/workflows/gcc-problems.yml]: same here

**Review by quentinDERORY — CHANGES_REQUESTED** [review]
Look at my comments and resolve linter issues.
Otherwise looking good.

File: .github/workflows/agw-workflow.yml (hunk 1 above)
Comment (ranj8work) [reply .github/workflows/agw-workflow.yml]: @quentinDERORY - Addressed the comments and pushed the changes

File: .github/workflows/agw-workflow.yml (hunk 2 above)
Comment (ranj8work) [reply .github/workflows/agw-workflow.yml]: @quentinDERORY - Addressed the comments and pushed the changes

File: .github/workflows/gcc-problems.yml (hunk 3 above)
Comment (ranj8work) [reply .github/workflows/gcc-problems.yml]: @quentinDERORY - Addressed the comments and pushed the changes

File: .github/workflows/gcc-problems.yml (hunk 4 above)
Comment (ranj8work) [reply .github/workflows/gcc-problems.yml]: @quentinDERORY - Addressed the comments and pushed the changes

**Comment by ranj8work** [comment]
@quentinDERORY - Thank you for the review comments. I addressed them and pushed the changes on to the branch.

**Comment by quentinDERORY** [comment]
> @quentinDERORY - Thank you for the review comments. I addressed them and pushed the changes on to the branch.

There are still some linting issues comments from the yaml linter.

File: .github/workflows/gcc-problems.yml [hunk 5]
```diff
@@ -20,23 +20,37 @@ on:
...
+    runs-on: ubuntu-latest
+    # Map a step output to a job output
+    outputs:
+      should_skip: ${{ steps.skip_check.outputs.should_skip }}
+    steps:
+      - id: skip_check
+        uses: fkirc/skip-duplicate-actions@master
+        with:
+          # Only run the main job for changes including the following paths
+          paths: '["orc8r/gateway/c/**", "lte/gateway/c/**"]'
```
Comment (quentinDERORY) [inline .github/workflows/gcc-problems.yml:30]: you did not deleted the path filter on the github action trigger in this file

Comment (ranj8work) [reply .github/workflows/gcc-problems.yml:30]: @quentinDERORY - addressed this comment and pushed the changes on to the branch. Please review.

**Review by quentinDERORY — APPROVED** [review]

**Comment by markjen** [comment]
I think this PR did not run all the actions to make sure the build would be green post-merge. 

For example, the markdown lint job started failing https://github.com/magma/magma/runs/3604837835

**Comment by quentinDERORY** [comment]
> I think this PR did not run all the actions to make sure the build would be green post-merge.
> 
> For example, the markdown lint job started failing https://github.com/magma/magma/runs/3604837835

This PR did not trigger the fail of this job. Here is the previous commit that was already failing: https://github.com/magma/magma/actions/runs/1240819128
Here is the list of run of the workflow for reference : https://github.com/magma/magma/actions/workflows/docs-workflow.yml?query=branch%3Amaster
And here is the commit that broke the workflow: https://github.com/magma/magma/actions/runs/1188190968

**Comment by markjen** [comment]
> > I think this PR did not run all the actions to make sure the build would be green post-merge.
> > For example, the markdown lint job started failing https://github.com/magma/magma/runs/3604837835
> 
> This PR did not trigger the fail of this job. Here is the previous commit that was already failing: https://github.com/magma/magma/actions/runs/1240819128
> Here is the list of run of the workflow for reference : https://github.com/magma/magma/actions/workflows/docs-workflow.yml?query=branch%3Amaster
> And here is the commit that broke the workflow: https://github.com/magma/magma/actions/runs/1188190968

Oops my bad, you're totally right! I've filed #9140 to track this

**Pull request merged**